### ISSUE TITLE
Extremely low loss but bad predictions on training set

### ISSUE THREAD

**Issue by LucasVandroux** [experiencor/basic-yolo-keras#157]
Hello @experiencor,

First, thanks a lot for letting us use your implementation of YOLOv2 with Keras. It saved me a lot of time.

However, after trying to train YOLO with the [KITTI Dataset](http://www.cvlibs.net/datasets/kitti/eval_object.php?obj_benchmark=2d) and the `inception3` architecture, I am running into a very bizarre problem: after less than 10 steps the loss on the training set is almost at 0 but when I use this architecture the predictions on the training set are actually quite bad.

I did my training the following way, 9 epochs with a learning rate of 1e-4 (stopped because of early stopping), then did 5 epochs more with a learning rate of 1e-5 (stopped again because of early stopping) and I get a loss equal to `0.0865` on my training set of `0.3186` on my validation set. This seems to be a case of overfitting for sure but even the detections on the training set at this point are not that good:

![image](https://user-images.githubusercontent.com/1956886/36531741-cdfba63a-17f9-11e8-883e-739d549b8a05.png)
![image](https://user-images.githubusercontent.com/1956886/36531778-e85c1c3a-17f9-11e8-8e76-3c66911a5f08.png)
![image](https://user-images.githubusercontent.com/1956886/36531819-008901f6-17fa-11e8-8331-dc7676f3ff54.png)
![image](https://user-images.githubusercontent.com/1956886/36531875-2cda5c32-17fa-11e8-94f6-e352916167b6.png)

For some predictions, the bounding box doesn't fit properly or the center seems to not be in the right cell.

Does anyone have an idea what I am probably doing wrong in the training of my CNN?

Thanks in advance for your help.

**Comment by LucasVandroux** [comment]
I found out what was my problem. First, I left the `train_times` at 10 and having a big dataset this obviously lead to overfitting and prevented an effective early stopping. Then the second problem is that I trained the network with different anchors than the ones I used for prediction which lead to the really inaccurate results I am showing in the previous message.

After correcting those two issues the results are pretty good.
